flixel/tile/FlxBaseTilemap.hx

```diff
@@ -1296,4 +1371,28 @@ abstract FlxTilemapDiagonalPolicy(Int)
 	 * Diagonal movement costs one more than orthogonal movement
 	 */
 	var WIDE = 2;
-}
\ No newline at end of file
+}
+
+enum FlxTilemapPathingMethod
+{
+	/**
+	 * Basic method is calculating distances using an array covering the whole map (was original solution)
+	*/
+	BASIC;
+	/**
+	 * Optimization of basic method where distances are only calculated in a zone around start/end tiles
+	 * - scale > 1.0 determines how this rectangular zone is grown around the rectangle formed by start-end points
+	 * - minHorizonSize is used if any the scaled zone dimension is too small
+	*/
+	BASIC_SCALED_HORIZON(scale:Float, minHorizonSize:Int);
+}
+
+typedef FlxTilemapZone =
```

Comment:
Do we really want to introduce more pooling? Might be overkill here, these objects should be much rarer compared to `FlxRect` or `FlxPoint`.